### ISSUE TITLE
[arm] v1.12-rc2 - ARM netlink namespaces broken (nil reference)

### ISSUE THREAD

**Issue by paralin** [docker/docker#24018]
Hi,

I'm running a custom build of v1.12-rc2 on Arm in an embedded linux environment.

The commands used to compile Docker can be found [here](https://github.com/paralin/buildroot/blob/master/package/docker-engine/docker-engine.mk). I'm presently trying to update to v1.12-rc2 from v1.11.1.

Here is the error Docker is getting when attempting to start:

```
May 21 22:36:04 u3 systemd[1]: Starting Docker Application Container Engine...
May 21 22:36:04 u3 dockerd[5308]: time="2016-05-21T22:36:04.625731116Z" level=info msg="New containerd process, pid: 5323"
May 21 22:36:05 u3 dockerd[5308]: time="2016-05-21T22:36:05.711054324Z" level=info msg="Graph migration to content-addressability took 0.00 seconds"
May 21 22:36:05 u3 dockerd[5308]: time="2016-05-21T22:36:05.716723949Z" level=warning msg="mountpoint for pids not found"
May 21 22:36:05 u3 dockerd[5308]: time="2016-05-21T22:36:05.720063074Z" level=info msg="Loading containers: start."
May 21 22:36:05 u3 dockerd[5308]: time="2016-05-21T22:36:05.913945908Z" level=info msg="Firewalld running: false"
May 21 22:36:06 u3 dockerd[5308]: time="2016-05-21T22:36:06.081978699Z" level=error msg="could not create netlink handle on initial namespace: protocol not supported"
May 21 22:36:06 u3 dockerd[5308]: time="2016-05-21T22:36:06.087732533Z" level=error msg="could not create netlink handle on initial namespace: protocol not supported"
May 21 22:36:06 u3 dockerd[5308]: panic: runtime error: invalid memory address or nil pointer dereference
May 21 22:36:06 u3 dockerd[5308]: [signal 0xb code=0x1 addr=0xc pc=0x989ff4]
May 21 22:36:06 u3 dockerd[5308]: goroutine 1 [running]:
May 21 22:36:06 u3 dockerd[5308]: panic(0x11e9158, 0x11e1e030)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/host/usr/lib/go/src/runtime/panic.go:481 +0x330
May 21 22:36:06 u3 dockerd[5308]: github.com/vishvananda/netlink.(*Handle).LinkByName(0x0, 0x1356aa8, 0x7, 0x0, 0x0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/vishvananda/netlink/link_linux.go:759 +0x$8
May 21 22:36:06 u3 dockerd[5308]: github.com/docker/libnetwork/netutils.ElectInterfaceAddresses(0x1356aa8, 0x7, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/docker/libnetwork/netutils/utils_linux.go$78 +0xac
May 21 22:36:06 u3 dockerd[5308]: github.com/docker/docker/daemon.initBridgeDriver(0xb4dcbcf8, 0x11e85680, 0x120fc700, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/docker/docker/daemon/daemon_unix.go:712 +$x4c0
May 21 22:36:06 u3 dockerd[5308]: github.com/docker/docker/daemon.(*Daemon).initNetworkController(0x11e01ee0, 0x120fc700, 0x121787e0, 0x0, 0x0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/docker/docker/daemon/daemon_unix.go:661 +$x5cc
May 21 22:36:06 u3 dockerd[5308]: github.com/docker/docker/daemon.(*Daemon).restore(0x11e01ee0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/docker/docker/daemon/daemon.go:223 +0xc98
May 21 22:36:06 u3 dockerd[5308]: github.com/docker/docker/daemon.NewDaemon(0x120fc700, 0xb5f32628, 0x12005620, 0xb5f32458, 0x12106310, 0x0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/vendor/src/github.com/docker/docker/daemon/daemon.go:596 +0x2aa$
May 21 22:36:06 u3 dockerd[5308]: main.(*DaemonCli).start(0x120f9de0, 0x0, 0x0)
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/cmd/dockerd/daemon.go:269 +0x165c
May 21 22:36:06 u3 dockerd[5308]: main.main()
May 21 22:36:06 u3 dockerd[5308]:         /skiff/workspaces/u/output/build/docker-engine-v1.12.0-rc2/cmd/dockerd/docker.go:68 +0x470
```

It looks like these are the two major errors:

```
could not create netlink handle on initial namespace: protocol not supported
```

and

```
mountpoint for pids not found
```

If this requires OS level changes to fix, I can do that. A bit interesting that this is suddenly broken as of v1.12, though.

**Comment by paralin** [comment]
The pids error is harmless and related to the kernel pids option not being enabled.

The other however, is much worse.

here is my check-config output: http://hastebin.com/raw/zivuhozaxo

**Comment by paralin** [comment]
Primary problem:

```
could not create netlink handle on initial namespace: protocol not supported
```

**Comment by paralin** [comment]
I have one Ethernet bridge in place by the way, br0, which contains my ethernet and wifi devices. Could be causing this problem.

**Comment by aboch** [comment]
This was fixed by #23859
Fix will be in 1.12-rc3

**Comment by paralin** [comment]
Awesome, thanks. I will build from HEAD in the meantime.
